packages/cli/api-importers/openapi/openapi-v2-importer/src/3.1/paths/RequestBodyConverter.ts

```diff
@@ -94,6 +92,176 @@ export class RequestBodyConverter extends AbstractConverter<
+            breadcrumbs: [...this.breadcrumbs, "schema"],
+            schemaOrReference: mediaTypeObject.schema,
+            schemaIdOverride: schemaId
+        });
+        const convertedSchema = schemaOrReferenceConverter.convert({ context, errorCollector });
+        if (convertedSchema == null) {
+            return undefined;
+        }
+
+        return convertedSchema;
+    }
+
+    private handleJsonOrFormContent = ({
+        contentType,
+        context,
+        errorCollector
+    }: {
+        contentType: string;
+        context: OpenAPIConverterContext3_1;
+        errorCollector: ErrorCollector;
+    }): RequestBodyConverter.Output | undefined => {
+        const schemaId = [...this.group, this.method, "Request"].join("_");
+        const convertedSchema = this.tryGetConvertedSchema({
+            schemaId,
+            contentType,
+            context,
+            errorCollector
+        });
+        if (convertedSchema == null) {
+            return undefined;
+        }
+
+        if (convertedSchema.schema?.shape.type === "object") {
+            const requestBody = HttpRequestBody.inlinedRequestBody({
+                contentType,
+                docs: this.requestBody.description,
+                name: context.casingsGenerator.generateName(schemaId),
+                extendedProperties: convertedSchema.schema?.shape.extendedProperties,
+                extends: convertedSchema.schema?.shape.extends,
+                properties: convertedSchema.schema?.shape.properties,
+                extraProperties: convertedSchema.schema?.shape.extraProperties
+            });
+
+            return {
+                requestBody,
+                inlinedTypes: Object.fromEntries(
+                    Object.entries(convertedSchema.inlinedTypes).filter(([key]) => key !== schemaId)
+                )
+            };
+        } else {
+            const requestBody = HttpRequestBody.reference({
+                contentType,
+                docs: this.requestBody.description,
+                requestBodyType: convertedSchema.type
+            });
+
+            return {
+                requestBody,
+                inlinedTypes: Object.fromEntries(
+                    Object.entries(convertedSchema.inlinedTypes).filter(([key]) => key !== schemaId)
+                )
+            };
+        }
+    };
+
+    private convertRequestBodyProperty({
+        context,
+        property,
+        contentType
+    }: {
+        context: OpenAPIConverterContext3_1;
+        property: ObjectProperty;
+        contentType: string;
+    }) {
+        if (context.isFile(property.valueType)) {
+            return FileUploadRequestProperty.file(
+                FileProperty.file({
+                    key: property.name,
+                    isOptional: false,
+                    contentType
+                })
+            );
+        } else if (
+            context.isOptional(property.valueType) &&
```

Comment:
if you make this a type guard in typescript, then you won't need to add the ts-expect-error

Reply:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
specifically the return type: `pet is Fish` in the example docs above